LSPS5/README.md

```diff
@@ -0,0 +1,512 @@
+    expected formats.
+  * MUST validate that the `POST` request body is a valid
+    [JSON-RPC 2.0 Notification Object][], which is the
+    webhook notification object.
+
+The notification delivery service, once it has received
+the timestamp, signature, and webhook notification object:
+
+* MUST check that the timestamp is within 10 minutes
+  (before or after) of its local time.
+* MUST remember the signature for at least 20 minutes,
+  and check that the current signature is not equal to
+  some remembered signature from a previous webhook
+  call.
+* MUST validate the signature against the message
+  template above, with the given timestamp and webhook
+  notification object.
+* MAY filter out particular notification methods by
+  its own policy.
+
+> **Rationale** The timestamp and signature checks are
+> intended to avoid replay attacks, where some third
+> party is able to copy the JSON object sent by the LSP
+> and replays it in order to frame the LSP for spamming
+> the client.
+
+The response to the webhook is unspecified.
+
+* For HTTPS webhooks (i.e. `POST`):
+  * The LSP MUST consider a `200 OK` response a
+    success, and MUST ignore the rest of the response,
+    including HTTP headers and the response body.
+  * If the response is not `200 OK`, the LSP SHOULD
+    log this as an unusual event, including the
+    HTTP headers and response body if any, and
+    MUST otherwise ignore the response.
+  * The LSP SHOULD NOT follow any redirection
+    responses.
+
+> **Rationale** As a notification, the user of the
+> client may ignore it, or disable notifications from
+> the client application entirely; thus, delivery of
+> the notification is never assured and it is pointless
+> to feed back anything to the LSP about whether the
+> notification was delivered to the notification
+> delivery service, the mobile OS server, or the client
+> mobile device.
+
+If the notification delivery service accepts the push
+notification, it MUST construct a suitable message to
+display to the user, and send that message, together
+with any required authentication and authorization, to
+the mobile OS server, for display as a push notification
+of the client.
+
+The client SHOULD process what is needed to
+react to the notification (e.g. connect to the LSP and
+accept the payment for a `lsps5.payment_incoming`
+notification) if the mobile environment allows it.
+
+#### Webhook Notification Methods
+
+The LSP MUST use one of the methods listed in this
+section for the webhook notification object.
+
+The LSP MUST NOT use any of these methods on the LSPS0
+interface; these notification methods are only for the
+webhook interface.
+
+Parameters are required unless otherwise stated.
+
+* `lsps5.webhook_registered` - The client has just
+  recently successfully called the `lsps5.set_webhook`
+  API.
+  No parameters `{}`.
+* `lsps5.payment_incoming` - The client has one or
+  more payments pending to be received.
+  No parameters `{}`.
```

Comment:
I know that we just removed the amountSat in here but I would like to have it back.
1) It is very useful for spam prevention. The NDS can decided if it wants to wake up the phone in case of a 1sat HTLC, or after the 99th 1sat HTLC within 5min.
2) The NDS does not need to forward this information to the 'mobile OS server'. Therefore it is not a real risk.

In general, the NDS needs a way to prevent spam for each method type and therefore needs enough data to actually make a decision.

Reply:
I requested to remove the `amount_sat`-field for privacy reasons.
At the time I didn't see a good reason to include it.

I believe spam protection is an actual use-case and don't object putting it back in.

Reply:
As noted in the February 1 meeting, the data can be faked, so seems to be minimally useful for spam prevention, so it seems that putting it back in is not useful.